selfdrive/ui/sidebar.cc

```diff
@@ -0,0 +1,255 @@
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include "ui.hpp"
+
+static void ui_draw_sidebar_background(UIState *s) {
```

Comment:
Can hasSidebar be passed in as an argument to all these functions? Then you only need to do the lookup once.